proxy/core/src/main/scala/io/cloudstate/proxy/AkkaActorRegisterFeature.scala

```diff
@@ -8,26 +8,25 @@ import scala.collection.JavaConverters._
 
 @AutomaticFeature
 final class AkkaActorRegisterFeature extends Feature {
-
+  val akkaActorClass = classOf[akka.actor.Actor]
   override def duringAnalysis(access: Feature.DuringAnalysisAccess) =
-    reachableSubtypes(access, access.findClassByName("akka.akka.Actor")).foreach { cls =>
-      for {
-        context <- getDeclaredField(cls, "context")
-        self <- getDeclaredField(cls, "self")
-      } {
-        RuntimeReflection.register(cls)
-        RuntimeReflection.register( /* finalIsWritable = */ true, context, self)
-        RuntimeReflection.register(cls.getDeclaredFields: _*)
-        RuntimeReflection.register(cls.getDeclaredConstructors: _*)
+    if (access.isReachable(akkaActorClass)) {
+      access.reachableSubtypes(akkaActorClass).iterator.asScala.filter(_ != null).foreach { subtype =>
+        if (subtype.getInterfaces.exists(_ == akkaActorClass)) {
+          for {
+            context <- getDeclaredField(subtype, "context")
+            self <- getDeclaredField(subtype, "self")
+          } {
+            RuntimeReflection.register(subtype)
+            RuntimeReflection.register( /* finalIsWritable = */ true, context, self)
+            //RuntimeReflection.register(subtype.getDeclaredFields: _*)
```

Comment:
@dwijnand I wasn't sure we needed this, but it can be debated.

Reply:
I kept it only because it's what the previous config did (IIRC).  But I'm in favour of registering the least necessary.

Reply:
If it works then we remove it :)